src/components/CreateMultipleTasksDialog.vue

```diff
@@ -0,0 +1,222 @@
+
+-->
+
+<template>
+	<NcModal class="task-selector" size="small" @close="() => {created ? cancel() : close()}">
+		<div v-if="!creating && !created" id="modal-inner">
+			<h3>{{ t('tasks', 'Create new tasks') }}</h3>
+
+			<p>{{ t('tasks', 'Create {numberOfTasks} tasks from pasted text', { numberOfTasks: tasksToCreate.numberOfTasks }) }}</p>
+
+			<div class="modal-buttons">
+				<NcButton @click="cancel">
+					{{ t('tasks', 'Cancel') }}
+				</NcButton>
+				<NcButton
+					type="primary"
+					@click="addTasks">
+					{{ t('tasks', 'Create tasks') }}
+				</NcButton>
+			</div>
+		</div>
+		<div v-else id="modal-inner">
+			<NcEmptyContent v-if="creating" key="creating" :description="t('tasks', 'Creating new tasks…')">
+				<template #icon>
+					<NcLoadingIcon />
+				</template>
+			</NcEmptyContent>
+			<NcEmptyContent v-else-if="created" key="created" :description="createdMessage">
+				<template #icon>
+					<Check />
+				</template>
+				<template #action>
+					<NcButton @click="close">
+						{{ t('tasks', 'Close') }}
+					</NcButton>
+				</template>
+			</NcEmptyContent>
+		</div>
+	</NcModal>
+</template>
+
+<script>
+
+import { translate as t } from '@nextcloud/l10n'
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcEmptyContent from '@nextcloud/vue/dist/Components/NcEmptyContent.js'
+import NcLoadingIcon from '@nextcloud/vue/dist/Components/NcLoadingIcon.js'
+import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
+
+import Check from 'vue-material-design-icons/Check.vue'
+import TextBoxOutline from 'vue-material-design-icons/TextBoxOutline.vue'
+import ViewHeadline from 'vue-material-design-icons/ViewHeadline.vue'
+
+import { mapActions } from 'vuex'
+
+export default {
+	name: 'CreateMultipleTasksDialog',
+	components: {
+		Check,
+		NcButton,
+		NcEmptyContent,
+		NcLoadingIcon,
+		NcModal,
+		TextBoxOutline,
+		ViewHeadline,
+	},
+	props: {
+		calendar: {
+			type: Object,
+			required: true
+		},
+		tasksToCreate: {
+			type: Object,
+			required: true
+		},
+		tasksAdditionalProperties: {
+			type: Object,
+			default: {}
+		},
+		rootTask: {
+			type: Object,
+			default: undefined
+		}
+	},
+	emits: ['cancel', 'close'],
+	data() {
+		return {
+			creating: false,
+			created: false,
+		}
+	},
+
+	computed: {
+		createdMessage() {
+			return t('tasks', '{numberOfTasks} tasks have been added to "{calendar}"', { numberOfTasks: this.tasksToCreate.numberOfTasks, calendar: this.calendar.displayName }, undefined, { sanitize: false, escape: false })
+		},
+	},
+
+	methods: {
+		...mapActions([
+			'createTask',
+		]),
+
+		t,
+
+		cancel() {
+			this.$emit('cancel')
+			this.$root.$emit('close')
+		},
+
+		close() {
+			this.$emit('close')
+			this.$root.$emit('close')
+		},
+
+		async addTasks() {
+			this.creating = true
+			Promise.all(this.tasksToCreate.tasks.map(task => this.addTaskWithParent(task, this.rootTask?.uid)))
```

Comment:
Could you please also use `await` here, like a few lines below?

Reply:
added